src/core/server/http/http_server.test.ts

```diff
@@ -992,6 +992,226 @@ describe('body options', () => {
   });
 });
 
+describe('timeout options', () => {
+  test('should accept a socket "timeout" which is 3 minutes in milliseconds, "300000" for a POST', async () => {
+    const { registerRouter, server: innerServer } = await server.setup(config);
+
+    const router = new Router('', logger, enhanceWithContext);
+    router.post(
+      {
+        path: '/',
+        validate: false,
+        options: { timeout: 300000 },
+      },
+      (context, req, res) => {
+        try {
+          return res.ok({ body: { timeout: req.route.options.timeout } });
+        } catch (err) {
+          return res.internalError({ body: err.message });
+        }
+      }
+    );
+    registerRouter(router);
+    await server.start();
+    await supertest(innerServer.listener).post('/').send({ test: 1 }).expect(200, {
+      timeout: 300000,
+    });
+  });
+
+  test('should accept a socket "timeout" which is 3 minutes in milliseconds, "300000" for a GET', async () => {
```

Comment:
Shouldn't it ignore the option for safe requests? https://github.com/elastic/kibana/pull/73103/files#diff-ab827d36e183b98981f29e3199353913R167

Reply:
Around those lines of code there are actually two different time outs happening. The one you linked to is the timeout associated with payloads for PUT, POST requests:

Payload one which will be undefined/non-existent unless it is a PUT/POST request:
```ts
             ? {
                  allow,
                  maxBytes,
                  output,
                  parse,
                  timeout: payloadTimeout,
                }
```

Then there is this second timeout which is around server and sockets in that file:
```ts
            timeout:
              timeout != null
                ? {
                    socket: timeout + 1, // Hapi server requires the socket to be greater than payload settings so we add 1 millisecond
                  }
                : undefined,
```

For PUT, POST, requests you have to ensure that the socket timeout is at least 1 millisecond above that of the payload timeout otherwise Hapi will throw an error. For GET, HEAD calls you cannot set the payload timeout or it will throw an error but you still want to set the socket timeout instead of using its default (which is 2 minutes).

So in short for keeping to a single timeout setting if you're using a PUT, POST you have to set both the payload and socket timeout and for GET, OPTIONS, you can only set the socket timeout.